docs/references/api.rst

```diff
@@ -227,11 +230,137 @@ This endpoint shows the installed Pro Client version.
+
+If you call this function outside of the boot process and would like the retry
+auto-attach functionality to actually start, you'll need to call something
+like ``systemctl start ubuntu-advantage.service``.
+
+- Introduced in Ubuntu Pro Client Version: ``27.12~``
+- Args:
+
+  - ``enable``: Optional list of services to enable after auto-attaching.
+  - ``enable_beta``: Optional list of beta services to enable after
+    auto-attaching.
+
+.. note::
+
+   If none of the lists are set, the services will be enabled based on the
+   contract definitions.
+
+.. tab-set::
+
+   .. tab-item:: Python API interaction
+      :sync: python
+
+      - Calling from Python code:
+
+        .. code-block:: python
+
+           from uaclient.api.u.pro.attach.auto.configure_retry_service.v1 import configure_retry_service, ConfigureRetryServiceOptions
+
+           options = ConfigureRetryServiceOptions(enable=["<service1>", "<service2>"], enable_beta=["<beta_service3>"])
+           result = configure_retry_service(options)
+
+      - Expected return object:
+
+        - ``uaclient.api.u.pro.attach.auto.configure_retry_service.v1.ConfigureRetryServiceResult``
+
+          No data present in the result.
+
+      - Raised exceptions:
+
+        - No exceptions raised by this endpoint.
+
+   .. tab-item:: CLI interaction
+      :sync: CLI
+
+      - Calling from the CLI:
+
+        - This endpoint currently has no CLI support. Only the Python-based
+          version is available.
+
+u.pro.attach.auto.full_auto_attach.v1
+=====================================
+
+This endpoint runs the whole auto-attach process on the system.
+
+- Introduced in Ubuntu Pro Client Version: ``27.11~``
+- Args:
+
+  - ``enable``: Optional list of services to enable after auto-attaching.
+  - ``enable_beta``: Optional list of beta services to enable after auto-attaching.
+
+.. note::
+
+   If none of the lists are set, the services will be enabled based on the
+   contract definitions.
+
+.. tab-set::
+
+   .. tab-item:: Python API interaction
+      :sync: python
+
+      - Calling from Python code:
+
+        .. code-block:: python
+
+           from uaclient.api.u.pro.attach.auto.full_auto_attach.v1 import full_auto_attach, FullAutoAttachOptions
+
+           options = FullAutoAttachOptions(enable=["<service1>", "<service2>"], enable_beta=["<beta_service3>"])
+           result = full_auto_attach(options)
+
+      - Expected return object:
+
+        - ``uaclient.api.u.pro.attach.auto.full_auto_attach.v1.FullAutoAttachResult``
+
+          No data present in the result.
+
+      - Raised exceptions
+
+        - ``AlreadyAttachedError``: Raised if running on a machine which is
+          already attached to a Pro subscription.
+        - ``AutoAttachDisabledError``: Raised if ``disable_auto_attach: true``
+          in ``uaclient.conf``.
+        - ``ConnectivityError``: Raised if it is not possible to connect to the
+          Contracts Server.
```

Comment:
More of a question than a suggestion, but is it "Contracts" server or "Contract" server? Because we have both shown in this page, and I'm not sure which is the correct one.